### PR TITLE
chore: return a struct with named and documented fields from `compact_persisting_batch`

### PR DESCRIPTION
**Pull request by alamb**
~Draft as it builds on https://github.com/influxdata/influxdb_iox/pull/5333~

# Rationale
As part of https://github.com/influxdata/conductor/issues/1068 I am reading the compactor / ingestion code and I found it hard to understand what was being returned from `compact_persisting_batch`

# Changes:
1. Return a `struct`
2. Add some doc comments.

### CODE REVIEW

File: ingester/src/compact.rs [hunk 1]
```diff
@@ -46,17 +46,38 @@ pub enum Error {
...
+/// Compact a given persisting batch into a [`CompactedStream`] or
+/// `None` if there is no data to compact.
 pub async fn compact_persisting_batch(
     time_provider: Arc<dyn TimeProvider>,
     executor: &Executor,
     namespace_id: i64,
     partition_info: &PartitionInfo,
     batch: Arc<PersistingBatch>,
-) -> Result<Option<(SendableRecordBatchStream, IoxMetadata, Option<SortKey>)>> {
+) -> Result<Option<CompactedStream>> {
```
Comment (alamb) [inline ingester/src/compact.rs:80]: this is the actual code change

File: ingester/src/compact.rs [hunk 2]
```diff
@@ -67,12 +88,14 @@ pub async fn compact_persisting_batch(
     let partition_key = &partition_info.partition.partition_key;
     let sort_key = partition_info.partition.sort_key();
 
-    // Get sort key from the catalog or compute it from cardinality. Save a new value for the
-    // catalog to store if necessary.
```
Comment (alamb) [inline ingester/src/compact.rs:71]: the catalog is not updated in this function

**Comment by alamb** [comment]
Ok, this PR is now properly ready for review now

**Comment by carols10cents** [comment]
Love it. I saw two teeensy tiny things to fix in the comments, so wafer-thin I felt bad about kicking them back to you, so I added a commit to take care of them ❤️

**Comment by alamb** [comment]
>  Love it. I saw two teeensy tiny things to fix in the comments; the changes were so wafer-thin I felt bad about kicking them back to you, so I added a commit to take care of them ❤️

Thank you!  https://github.com/influxdata/influxdb_iox/pull/5346/commits/ce76c4dc0483f623130248b6bea65f7f21b8b059 looks great 👍

**Pull request merged**